source-code/core2/src/messages/query.ts

```diff
@@ -1,15 +1,58 @@
+import { createInlang } from "../app/createInlang.js"
 import type { Message } from "./schema.js"
 
-// TODO
-// 1. Query builder?
-// 2. Filtering etc how?
-//
+type UniqueFilterType = {
+	id: string
+	languageTag: string
+	//variant: string -> in the future
+}
+type FilterType = {
+	id?: string
+	languageTag?: string
+	//variant?: string -> in the future
+}
+
 // inspiration: Prisma CRUD https://www.prisma.io/docs/concepts/components/prisma-client/crud
 export type MessagesQueryApi_1 = {
-	create: (message: Message) => void
-	get: () => Message[]
-	update: () => void
-	delete: () => void
+	create: (args: Message) => Message
+	get: (args: { where: UniqueFilterType }) => [Message, Error]
+	getMany: (args: { where: FilterType }) => [Message[], Error]
+	update: (args: Message) => Message
+	updateMany: (args: Message[]) => [Message[], Error]
+	upsert: (args: Message) => [Message, Error]
+	upsertMany: (args: Message[]) => [Message[], Error]
```

Comment:
From call with @NiklasBuchfink this will be a super convenient function for the editor. Just put the local changes (array of messages) in the function.

Reply:
Yes, agree!